optuna/_transform.py

```diff
@@ -114,6 +121,13 @@ def transform(self, params: Dict[str, Any]) -> numpy.ndarray:
                 )
                 bound_idx += 1
 
+        if self._transform_0_1:
+            single_mask = self._raw_bounds[:, 0] == self._raw_bounds[:, 1]
+            trans_params[single_mask] = 0.5
+            trans_params[~single_mask] = (trans_params[~single_mask] - self._raw_bounds[:, 0]) / (
+                self._raw_bounds[:, 1] - self._raw_bounds[:, 0]
+            )
```

Comment:
I noticed that we need to filter with `~single_mask` like this, otherwise `ValueError: operands could not be broadcast together with shapes (7,) (9,)` exception can be raised.

```suggestion
            trans_params[~single_mask] = (
                trans_params[~single_mask] - self._raw_bounds[~single_mask, 0]
            ) / (self._raw_bounds[~single_mask, 1] - self._raw_bounds[~single_mask, 0])
```

Reply:
Thanks for pointing that out!
I'll apply the suggestion.